core/src/Http/HttpRequest.cs

```diff
@@ -74,16 +80,39 @@ private static HttpRequestMessage CreateRequestMessage(Uri endpoint, Dictionary<
             return requestMessage;
         }
 
+        private static async Task<HttpContent> CloneContent(HttpContent content)
+        {
+            var temp = new System.IO.MemoryStream();
+            await content.CopyToAsync(temp).ConfigureAwait(false);
+            temp.Position = 0;
+            var clone = new StreamContent(temp);
+            if (content.Headers != null)
+            {
+                foreach (var h in content.Headers)
+                {
+                    clone.Headers.Add(h.Key, h.Value);
+                }
+            }
+            return clone;
+        }
+
         private static async Task<HttpResponse> ExecuteWithRetry(Uri endpoint, Dictionary<string, string> headers,
-            Dictionary<string, string> bodyParameters, HttpMethod method,
+            HttpContent body, HttpMethod method,
             RequestContext requestContext, bool retry = true)
         {
             Exception toThrow = null;
             bool isRetryable = false;
             HttpResponse response = null;
             try
             {
-                response = await Execute(endpoint, headers, bodyParameters, method);
+                HttpContent content = null;
```

Comment:
>content [](start = 28, length = 7)

since this is only the body element should it then rather be called clonedBody or similar?